core/src/main/java/com/oath/oak/EntrySet.java

```diff
@@ -0,0 +1,951 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import sun.misc.Unsafe;
+
+import java.nio.ByteBuffer;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+import static com.oath.oak.ValueUtils.INVALID_VERSION;
+import static com.oath.oak.ValueUtils.ValueResult.FALSE;
+import static com.oath.oak.ValueUtils.ValueResult.TRUE;
+
+/* EntrySet keeps a set of entries. Entry is reference to key and value, both located off-heap.
```

Comment:
maybe elaborate on entry - it is not a class but represented as a consecutive integers in an array of ints.

Reply:
added big comment